src/app/shared/fabric8-ui-http.service.ts

```diff
@@ -29,6 +64,39 @@ export class Fabric8UIHttpService extends Http {
 
   request(url: string | Request, options?: RequestOptionsArgs): Observable<Response> {
     let urlStr = (typeof url === 'string' ? url : url.url);
+
+    if (isGetMethod(url, options)) {
+
+      const cacheKey = createCacheKey(url, options);
+      if (cacheKey in this.pendingRequests) {
+        // return the cached result
+        return this.pendingRequests[cacheKey];
+      }
+
+      const result = this.makeRequest(urlStr, url, options).do(() => {
```

Comment:
If the request produces an error, I think this timeout and cache clearing won't happen. Is this intentional? Maybe this should be `finally` rather than `do`?